tests/core/v5_1/test_v51_rpc_handlers.py

```diff
@@ -135,24 +153,86 @@ async def test_v51_rpc_ping_web3(make_request, bob_node_id_param_w3, alice, bob,
 
 
 @pytest.mark.trio
-async def test_v51_rpc_send_ping(make_request, bob_node_id_param, bob_network):
-    async with bob_network.client.dispatcher.subscribe(PingMessage) as subscription:
-        response = await make_request("discv5_sendPing", [bob_node_id_param])
-        with trio.fail_after(2):
-            request = await subscription.receive()
-        assert encode_hex(request.message.request_id) == response["request_id"]
+async def test_v51_rpc_get_enr(make_request, bob, bob_node_id_param):
+    bob_response = await make_request("discv5_getENR", [bob_node_id_param])
+    assert bob_response["enr_repr"] == repr(bob.enr)
 
 
 @pytest.mark.trio
-async def test_v51_rpc_send_ping_invalid_node_id(make_request, invalid_node_id):
+async def test_v51_rpc_get_enr_with_unseen_node_id(make_request, new_enr):
     with pytest.raises(Exception, match="'error':"):
-        await make_request("discv5_sendPing", [invalid_node_id])
+        await make_request("discv5_getENR", [repr(new_enr)])
 
 
 @pytest.mark.trio
-async def test_v51_rpc_send_ping_missing_node_id(make_request):
+async def test_v51_rpc_get_enr_web3(make_request, bob, bob_node_id_param_w3, w3):
+    response = await trio.to_thread.run_sync(w3.discv5.get_enr, bob_node_id_param_w3)
```

Comment:
What happens if the ENR isn't in the database?  (I think that the `w3` API should raise an exception in this case similar to what we did with `w3.eth.getTransaction` a while back so that we didn't have to do return value checking.

Reply:
Yup, it'll raise an exception, I've added a test case to show this.